### PR TITLE
Use tgui-styles

### PR DESCRIPTION
**Pull request by mordent-goonstation**
[UI][INTERNAL][CODE QUALITY]

## About the PR 

Uses https://github.com/tgstation/tgui-styles, removing our own versions of those styles.

There are a couple of Goonstation changes I need to either get merged into tgui-styles itself or put back in some capacity before this can be merged.

## Why's this needed? 

Keeps us more in-line with upstream tgui, helping us stay up-to-date.

## Port Upstream/Extract

- [ ] `.fas`/`.far` <-> `.fa-solid`/`.fa-regular`
- [ ] `.Modal--fitted`
- [x] `.Button--ellipsis .Button__content`
- [ ] `.Dimmer--full`

### CODE REVIEW

File: tgui/packages/tgui/styles/components/Modal.scss [hunk 1]
```diff
@@ -1,19 +0,0 @@
...
-.Modal {
-  background-color: $background-color;
-  max-width: calc(100% - 1rem);
-  padding: 1rem;
-}
-
-// |GOONSTATION-ADD|
-.Modal--fitted {
-  padding: 0;
-}
```
Comment (mordent-goonstation) [inline tgui/packages/tgui/styles/components/Modal.scss:19]: To port/verify is ported upstream

File: tgui/packages/tgui/styles/components/Section.scss [hunk 2]
```diff
@@ -1,162 +0,0 @@
...
-//   flex-grow: 1;
-// }
-
-// .content__noTopPadding {
-//   padding-top: 0;
-// }
-
-// .content__stretchContents {
-//   height: calc(100% - 3rem);
-// }
```
Comment (mordent-goonstation) [inline tgui/packages/tgui/styles/components/Section.scss:162]: Fixed upstream already.

File: tgui/packages/tgui/styles/main.scss [hunk 3]
```diff
@@ -19,40 +20,9 @@
...
-@include meta.load-css('./components/Dialog.scss');
-@include meta.load-css('./components/Dimmer.scss');
-@include meta.load-css('./components/Divider.scss');
-@include meta.load-css('./components/Dropdown.scss');
-@include meta.load-css('./components/Flex.scss');
-@include meta.load-css('./components/Icon.scss');
-// |GOONSTATION-CHANGE| This component's stylesheet includes styles that apply to other things,
-// and the component is not used by Goonstation; including here for completeness, and a note that
-// until this is fixed including *all* of tgui-styles may break a bunch of our styles
-// @include meta.load-css('./components/ImageButton.scss');
```
Comment (mordent-goonstation) [inline tgui/packages/tgui/styles/main.scss:35]: Fixed upstream, we do not consume this component currently (but its styles would be included if this was still a problem).

File: tgui/packages/tgui/styles/components/Button.scss [hunk 4]
```diff
@@ -1,188 +0,0 @@
...
-  -ms-user-select: none;
-
-  &:last-child {
-    margin-right: 0;
-    margin-bottom: 0;
-  }
-
-  .fa,
-  .fa-solid, // |GOONSTATION-CHANGE|
-  .fa-regular // |GOONSTATION-CHANGE|
```
Comment (mordent-goonstation) [inline tgui/packages/tgui/styles/components/Button.scss:65]: Verify these aren't needed, or possibly port our usages to `fas`/`far`.

File: tgui/packages/tgui/styles/components/Button.scss [hunk 5]
```diff
@@ -1,188 +0,0 @@
...
-  line-height: base.em(16px);
-  height: base.em(22px);
-  padding: 0.2rem 0.5rem;
-}
-
-.Button--hasContent {
-  // Add a margin to the icon to keep it separate from the text
-  .fa,
-  .fa-solid, // |GOONSTATION-CHANGE|
-  .fa-regular // |GOONSTATION-CHANGE|
```
Comment (mordent-goonstation) [inline tgui/packages/tgui/styles/components/Button.scss:84]: Verify these aren't needed, or possibly port our usages to `fas`/`far`.

File: tgui/packages/tgui/styles/components/Button.scss [hunk 6]
```diff
@@ -1,188 +0,0 @@
...
-  {
-    margin-right: 0.25em;
-  }
-}
-
-.Button--hasContent.Button--iconPosition--right {
-  .fa,
-  .fa-solid, // |GOONSTATION-CHANGE|
-  .fa-regular // |GOONSTATION-CHANGE|
```
Comment (mordent-goonstation) [inline tgui/packages/tgui/styles/components/Button.scss:93]: Verify these aren't needed, or possibly port our usages to `fas`/`far`.

File: tgui/packages/tgui/styles/components/Button.scss [hunk 7]
```diff
@@ -1,188 +0,0 @@
...
-  text-overflow: ellipsis;
-  overflow: hidden;
-}
-
-// |GOONSTATION-ADD|
-// ellipsis has to be on the children, it can't work otherwise.
-.Button--ellipsis .Button__content {
-  overflow: hidden;
-  text-overflow: ellipsis;
-}
```
Comment (mordent-goonstation) [inline tgui/packages/tgui/styles/components/Button.scss:111]: Check upstream.

File: tgui/packages/tgui/styles/components/Dimmer.scss [hunk 8]
```diff
@@ -1,28 +0,0 @@
...
-  // Dim everything around it
-  background-color: rgba(0, 0, 0, $background-dimness);
-  z-index: 1;
-}
-
-// |GOONSTATION-ADD|
-.Dimmer--full {
-  position: fixed;
-  top: 32px;
-}
```
Comment (mordent-goonstation) [inline tgui/packages/tgui/styles/components/Dimmer.scss:28]: Port upstream/verify ported upstream.

File: tgui/packages/tgui/styles/components/Button.scss (hunk 7 above)
Comment (Garash2k) [reply tgui/packages/tgui/styles/components/Button.scss:111]: Shouldn't be needed anymore as they fixed it in https://github.com/tgstation/tgstation/pull/80194

Comment (mordent-goonstation) [reply tgui/packages/tgui/styles/components/Button.scss:111]: Verified that change got through to tgui-styles, seems fine.

**Pull request closed without merging**